src/cmd/ksh93/sh/io.c

```diff
@@ -1881,6 +1881,24 @@ void sh_vexsave(Shell_t *shp,int fn,int fd,Spawnvex_f vexfun, void *arg)
 	}
 }
 
+// Return the lowest numbered fd that is equal to or greater than the requested
+// `min_fd` and which is not currently in use.
+int sh_get_unused_fd(Shell_t* shp, int min_fd) {
+	int fd;
+
+	while (true) {
+	    if (fcntl(min_fd, F_GETFD) == -1) {
+		for(fd = 0; fd < shp->topfd; fd++) {
```

Comment:
Fix indentation.

Reply:
The weird indentation is because of the inconsistent indentation of the code which confuses the Vim plugin I use that attempts to detect and set the indentation behavior. Once I've reformatted the code to have a consistent format this will no longer be an issue.